### PR TITLE
DAOS-12728 test: Fix dictionary string conversion

### PR DESCRIPTION
**Pull request by phender**
Adding general_utils.py dict_to_list() and dict_to_str() methods to ensure items are converted to strings before being joined.

Skip-unit-tests: true
Test-tag: test_macsio test_query_attr test_dfuse_daos_build_data test_soak_smoke

Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/general_utils.py [hunk 1]
```diff
@@ -773,24 +773,48 @@ def check_pool_files(log, hosts, uuid):
...
+        joiner (str, optional): string to use to join each key and value. Defaults to "=".
+
+    Returns:
+        list: a list of joined dictionary key and value strings
+
+    """
+    return [list_to_str(items, joiner) for items in value.items()]
+
+
+def dict_to_str(value, joiner=", ", items_joiner="="):
```
Comment (daltonbohning) [inline src/tests/ftest/util/general_utils.py:804]: Seems all uses of `dict_to_str` pass `joiner=','` or `joiner=:`. So does the default of `", "` still make sense?

**Review by daltonbohning — APPROVED** [review]
This will conflict with my #11534, which should be ready to land soon. Just waiting for the last HW stage to finish testing

**Review by daltonbohning — APPROVED** [review]

File: src/tests/ftest/util/general_utils.py (hunk 1 above)
Comment (phender) [reply src/tests/ftest/util/general_utils.py:804]: It's almost 50/50 currently. Since everything is passing I'm inclined to leave it the way it is.

**Review by mjean308 — APPROVED** [review]

File: src/tests/ftest/util/general_utils.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/util/general_utils.py:804]: How can it be 50/50 when there are three options above?

File: src/tests/ftest/util/general_utils.py [hunk 2]
```diff
@@ -773,24 +773,48 @@ def check_pool_files(log, hosts, uuid):
...
 
 
-def convert_list(value, separator=","):
-    """Convert a list into a separator-separated string of its items.
+def list_to_str(value, joiner=","):
+    """Convert a list to a string by joining its items.
+
+    Args:
+        value (list): list to convert to a string
+        joiner (str, optional): _description_. Defaults to ",".
```
Comment (ashleypittman) [inline src/tests/ftest/util/general_utils.py:781]: Well, don't keep us in suspense, what is the description?

I assume you're using a tool to pre-populate these annotations, can you share the details please?

Comment (daltonbohning) [reply src/tests/ftest/util/general_utils.py:781]: @ashleypittman Okay if I update this description in a followup PR? I need to rebase one of my PRs on these changes

Comment (daltonbohning) [reply src/tests/ftest/util/general_utils.py:781]: autoDocstring for VS Code: https://marketplace.visualstudio.com/items?itemName=njpwerner.autodocstring

Comment (ashleypittman) [reply src/tests/ftest/util/general_utils.py:781]: That's fine.  Could you do a quick "git grep _description_" and see if you can fill in any of the others as well please.

Comment (daltonbohning) [reply src/tests/ftest/util/general_utils.py:781]: Sure, I'll search all ftest for `_description_`, etc.

**Comment by daltonbohning** [comment]
@ashleypittman Otherwise okay to land this one?

File: src/tests/ftest/util/general_utils.py (hunk 1 above)
Comment (ashleypittman) [reply src/tests/ftest/util/general_utils.py:804]: What's the resolution here?  Are ", " and "," both valid and considered different?

Comment (ashleypittman) [reply src/tests/ftest/util/general_utils.py:804]: Fine to land with me.

File: src/tests/ftest/util/general_utils.py (hunk 2 above)
Comment (daltonbohning) [reply src/tests/ftest/util/general_utils.py:781]: FYI #11607

**Pull request merged**